linera-chain/src/chain.rs

```diff
@@ -494,14 +495,28 @@ where
         let balance = self.execution_state.system.balance.get_mut();
 
         balance.try_add_assign(credit)?;
-        Self::sub_assign_fees(balance, pricing.certificate_price())?;
-        Self::sub_assign_fees(balance, pricing.storage_price(&block.incoming_messages)?)?;
-        Self::sub_assign_fees(balance, pricing.storage_price(&block.operations)?)?;
+        sub_assign_fees(balance, policy.certificate_price())?;
```

Comment:
Removing Self:: may or may not be an improvement.

Reply:
It may or it may not, I agree, but I see this as rather not essential.